eng/scripts/Get-BinarySizes.ps1

```diff
@@ -0,0 +1,179 @@
+[CmdletBinding()]
+param(
+    [Parameter()]
+    [string] $BuildDirectory = "$PSScriptRoot/../../../build/",
+
+    [Parameter()]
+    [string] $ServiceDirectory = "*",
+
+    [Parameter()]
+    [string] $OsVMImage,
+
+    [Parameter()]
+    [string] $CmakeArgs,
+
+    [Parameter()]
+    [string] $CmakeEnvArg,
+
+    [Parameter()]
+    [string] $BuildArgs,
+
+    [Parameter()]
+    [string] $Job,
+
+    [Parameter()]
+    [string] $BuildReason,
+
+    [Parameter()]
+    [string] $SourceBranch,
+
+    [Parameter()]
+    [switch] $CI = ($null -ne $env:SYSTEM_TEAMPROJECTID),
+
+    [Parameter()]
+    [hashtable]$ExtraLabels = @{}
+)
+
+. $PSScriptRoot/../common/scripts/common.ps1
+
+function setEnvVar($key, $value) {
+    Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$value"
+}
+
+function getTargetOs {
+    if ($OsVMImage.StartsWith('macos')) {
```

Comment:
If we're going to start reporting OS image metrics from multiple places, I think it makes sense to think deeper about standardization. We could grab the OS value from the distro itself, e.g. `lsb_release -a` or perhaps the virtual environments repo embeds information about the image somewhere into the filesystem? Otherwise we're likely going to end up with multiple metric dimensions and values that slightly differ (e.g. ubuntu-18.04 and MMSUbuntu18.04, etc., etc.).

Reply:
This is another area where I think it makes sense to have a standardized "environment metrics" script that we can import from eng/common along with a base metric object builder that gets stuff like timestamps formatted consistently.

```
$extraMetrics = @{ key1 = value1; key2 = value2 }
$metric = BuildMetric $extraMetrics  # from eng/common
LogMetric $metric  # from eng/common
```

Reply:
Many of these metrics are specific to variables specified in the matrix (e.g. `CMAKE_BUILD_TYPE`) ... 

Even in the case of `CMAKE_BUILD_TYPE` the configuration information is not definitive because some build toolchains require that build configuration be specified during the build process instead of during the generate process (in that case we use `BuildArgs`)

[cmake build configurations](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-configurations) 

Virtual environments hosted in an Azure datacenter do have the [metadata API](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/instance-metadata-service?tabs=windows) ... but MacOS machines do not have it (as of last time I checked circa 2019-2020).

Reply:
Back to your point, yes... we should get OS information from the OS... we can also get relevant build information from the parameters.